src/registrar/models/domain.py

```diff
@@ -1293,6 +1293,51 @@ def _fetch_cache(self, fetch_hosts=False, fetch_contacts=False):
         except RegistryError as e:
             logger.error(e)
 
+        except TransitionNotAllowed as err:
+            # Fixes a bug with _fetch_cache trying to create
+            # a deleted domain, as cache gets cleared on delete.
+            # Instead, serve what we have locally.
+            if self.state == self.State.DELETED:
+                logger.warning("Attempted to create a deleted domain")
+                data = self._cache
+                choices = PublicContact.ContactTypeChoices
+                contacts_dict = {
+                    choices.ADMINISTRATIVE: None,
+                    choices.SECURITY: None,
+                    choices.TECHNICAL: None,
+                }
+                registrant_id = ...
+                existing_contacts = PublicContact.objects.filter(domain=self)
+                if existing_contacts.count() > 0:
+                    for choice in contacts_dict:
+                        contacts_dict[choice] = existing_contacts.get(
+                            contact_type=choice
+                        ).registry_id
+                    # Edge case for registrant
+                    registrant = PublicContact.ContactTypeChoices.REGISTRANT
+                    registrant_id = existing_contacts.get(
+                        contact_type=registrant
+                    ).registry_id
```

Comment:
For some background on this:
When clicking the "view" button on a deleted domain on the registrar / .gov side, it'll error out (500) if cache gets cleared for that object - meaning you can't view deleted domains.

The reason for this is that, because cache is empty, it calls _fetch_cache. _fetch_cache will then call `_get_or_create_domain()` which inevitably invokes the function `pendingCreate()` (because it was deleted from the registry). Since pendingCreate violates the FSM rule of unknown => dns_needed (deleted => dns_needed), it fails. As a test, overriding that causes other issues with how the page is currently setup because it still has the save buttons, etc.

I chose to handle this in an exception as it was minimally code invasive, but there are multiple solutions to it, so I have no problem changing my approach.

As for your comment, I think that is a good idea generally, but not in the context of this particular solution. I personally would opt instead to outright not show the security contact page / info, plus the other stuff like nameservers etc if they will always be None